src/item.cpp

```diff
@@ -2371,6 +2376,35 @@ bool item::has_flag( const std::string &f ) const
     return ret;
 }
 
+bool item::has_property (const std::string& prop) const {
+   return type->properties.find(prop) != type->properties.end();
+}
+
+std::string item::get_property (const std::string &prop, const std::string& def) const
```

Comment:
> Also the API where you determine the return value based on the default value is very odd, please just distinguish between them with names, e.g. get_property_int()

But why? It's simple, it's short, it's not redundant, it avoids parsing the constant default value. If given a default value of type X, the caller apparently expects to get a value of type X. Otherwise, why would the caller give a default value that would need to be converted to another type when returned.

What about a template version:
```C++
template<typename R>
R get_property(const std::string &name, R default_value = R()) {
    return ...
}

int g = get_property<int>("");
int f = get_property<int>("", 'C');
int h = get_property("", 0);
```